File: sample-objc/ARGEARSample/ARGearHelpers/BulgeManager.m
//
//  BulgeManager.m
//  ARGEARSample
//
//  Created by Jihye on 24/10/2019.
//  Copyright © 2019 Seerslab. All rights reserved.
//

#import "BulgeManager.h"

@implementation BulgeManager

+ (BulgeManager *)shared {
    static dispatch_once_t pred;
    static BulgeManager *sharedInstance = nil;
    dispatch_once(&pred, ^{
        sharedInstance = [[BulgeManager alloc] init];
    });
    return sharedInstance;
}

- (instancetype)init {
    self = [super init];
    if (self) {
        _mode = ARGContentItemBulgeNONE;
    }
    return self;
}

- (void)off {
    [[_session contents] setBulge:ARGContentItemBulgeNONE];
    
    _mode = ARGContentItemBulgeNONE;
}

- (void)setFunMode:(ARGContentItemBulge)mode {
    [[_session contents] setBulge:mode];
    
    _mode = mode;
}

@end

File: sample-swift/ARGearSample/ARGearHelpers/BulgeManager.swift
//
//  BulgeManager.swift
//  ARGearSample
//
//  Created by Jihye on 2020/02/10.
//  Copyright © 2020 Seerslab. All rights reserved.
//

import Foundation
import ARGear

class BulgeManager {
    static let shared = BulgeManager()
    
    var argSession: ARGSession?
    var mode: ARGContentItemBulge = .NONE
    
    init() {
    }
    
    public func off() {
        guard let session = self.argSession, let contents = session.contents
            else { return }
        
        contents.setBulge(.NONE)
        
        mode = .NONE
    }
    
    public func setFunMode(_ mode: ARGContentItemBulge) {
        guard let session = self.argSession, let contents = session.contents
            else { return }
        
        contents.setBulge(mode)
        
        self.mode = mode
    }
}
